modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsConsumer.java

```diff
@@ -134,18 +134,21 @@ public boolean ack() {
         return result;
     }
 
-    public boolean cleanup() {
+    public boolean cleanup() throws SynapseException {
         if (logger.isDebugEnabled()) {
             logger.debug(getId() + " cleaning up...");
         }
-        boolean result =  store.cleanup(connection, session, true);
-        if (result) {
+        try {
+            store.cleanup(connection, session);
+            return true;
+        } catch (JMSException e) {
+            throw new SynapseException("Error while connecting to store to close created connections. JMS provider " +
+                    "might not be accessible" + store.getName(), e);
```

Comment:
Operator should precede the line shall we fix this?

Reply:
Fixed.